wasm/wat/types.go

```diff
@@ -0,0 +1,83 @@
+package wat
+
+import "fmt"
+
+// module corresponds to the text format of a WebAssembly module, and is an intermediate representation prior to
+// wasm.Module. This is primarily needed to resolve symbolic indexes like "$main" to raw numeric ones.
+//
+// Note: nothing is required per specification. Ex `(module)` is valid!
+//
+// See https://www.w3.org/TR/wasm-core-1/#functions%E2%91%A7
+type module struct {
+	// name is optional and starts with '$'. For example, "$test".
+	// See https://www.w3.org/TR/wasm-core-1/#modules%E2%91%A0%E2%91%A2
+	//
+	// Note: The name may also be stored in the wasm.Module CustomSection under the key "name" subsection 0.
+	// See https://www.w3.org/TR/wasm-core-1/#binary-namesec
+	name string
+
+	// importFuncs are imports describing functions "(import... (func...))" added in insertion order.
+	importFuncs []*importFunc
+
+	// startFunction is the function to call during wasm.Store Instantiate. The value is a importFunc.name, such as
+	// "$main", or its equivalent raw numeric index, such as "2".
+	//
+	// Note: When in raw numeric form, this is relative to imports. See wasm.Module StartSection for more.
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#start-function%E2%91%A4
+	startFunction string
+}
+
+// importFunc corresponds to the text format of a WebAssembly function import.
+//
+// Note: nothing is required per specification. Ex `(import "" "" (func))` is valid!
```

Comment:
Didn't know that!

Reply:
not sure it is intended to be so vague, but heh the grammar allows it